src/mavsdk/core/fs_utils.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "fs_utils.h"
+#include "log.h"
+#include <fstream>
+#include <random>
+
+#if defined(LINUX) || defined(__APPLE__)
+#include <pwd.h>
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#ifdef _WIN32
+#include <windows.h>
+#include <shlobj.h>
+
+static std::optional<std::filesystem::path> get_known_windows_path(REFKNOWNFOLDERID folderId)
+{
+    std::filesystem::path path;
+    PWSTR path_tmp;
+
+    /* Attempt to get user's AppData folder
+     *
+     * Microsoft Docs:
+     * https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath
+     * https://learn.microsoft.com/en-us/windows/win32/shell/knownfolderid
+     */
+    auto get_folder_path_ret = SHGetKnownFolderPath(folderId, 0, nullptr, &path_tmp);
+
+    /* Error check */
+    if (get_folder_path_ret != S_OK) {
+        CoTaskMemFree(path_tmp);
+        return std::nullopt;
+    }
+
+    /* Convert the Windows path type to a C++ path */
+    path = path_tmp;
+
+    CoTaskMemFree(path_tmp);
+    return path;
+}
+#endif
+
+namespace mavsdk {
+
+std::optional<std::filesystem::path> get_cache_directory()
+{
+    // Windows: %LOCALAPPDATA%
+    // Android: /data/data/<package>/cache
+    // Linux: ~/.cache
+    // macOS: ~/Library/Caches
+#if defined(_WIN32)
+    auto path = get_known_windows_path(FOLDERID_LocalAppData);
+    if (path) {
+        return path->append("mavsdk_cache");
+    }
+#elif defined(__ANDROID__)
```

Comment:
And `#elif defined(ANDROID)`

Reply:
https://github.com/mavlink/MAVSDK/pull/2272/commits/998ed155ccdcc02aff4dc236cc096e4455dedb93